### PR TITLE
fix: scene sorting was being performed ten times in the same frame in some cases

### PR DESCRIPTION
**Pull request by BrianAmadori**
This issue was introducing 100 ms hiccups.

### CODE REVIEW

File: unity-client/Assets/Scripts/MainScripts/DCL/Controllers/Scene/SceneController.cs [hunk 1]
```diff
@@ -469,7 +474,8 @@ public void LoadParcelScenes(string decentralandSceneJSON)
 
             MessagingControllersManager.i.ForceEnqueueToGlobal(MessagingBusId.INIT, queuedMessage);
 
-            TrySortScenesByDistance(forceSort: true);
+            sceneSortDirty = true;
+            TrySortScenesByDistance();
```
Comment (eordano) [inline unity-client/Assets/Scripts/MainScripts/DCL/Controllers/Scene/SceneController.cs]: I feel like we can keep the `true` here, what do you think?

```suggestion
            TrySortScenesByDistance(true);
```

**Review by eordano — APPROVED** [review]
Looks good to me. Minor consideration

**Review by AjimenezDCL — APPROVED** [review]
I think we could tackle down the issue in an easier way.

Surely `SceneController.LoadParcelScenes` is the one causing the hiccups. Do we really need that method to call `TrySortScenesByDistance`? I think we can remove the call (and therefore remove the `forceSort` param) and only call `TrySortScenesByDistance` in the update using the existent `sceneSortDirty` logic. The biggest issue could be 1 frame delay between detecting the player's new position.

I'm approving nonetheless,

**Review by pravusjif — APPROVED** [review]

**Comment by pravusjif** [comment]
> I think we could tackle down the issue in an easier way.
> 
> Surely `SceneController.LoadParcelScenes` is the one causing the hiccups. Do we really need that method to call `TrySortScenesByDistance`? I think we can remove the call (and therefore remove the `forceSort` param) and only call `TrySortScenesByDistance` in the update using the existent `sceneSortDirty` logic. The biggest issue could be 1 frame delay between detecting the player's new position.
> 
> I'm approving nonetheless,

this may be worth checking out

**Pull request merged**